cmd/dockerd/docker_windows.go

```diff
@@ -1,19 +1,29 @@
 package main
 
 import (
+	"context"
+	"fmt"
 	"io"
-	"path/filepath"
+	"os"
 
 	"github.com/Microsoft/go-winio/pkg/etwlogrus"
 	"github.com/containerd/log"
 )
 
-func runDaemon(opts *daemonOptions) error {
-	daemonCli := NewDaemonCli()
+func runDaemon(ctx context.Context, opts *daemonOptions) error {
+	cli, err := newDaemonCLI(opts)
+	if err != nil {
+		return err
+	}
+	if opts.Validate {
+		// If config wasn't OK we wouldn't have made it this far.
+		_, _ = fmt.Fprintln(os.Stderr, "configuration OK")
+		return nil
+	}
```

Comment:
This block of code is now line-for-line identical between both platforms. I smell a refactoring opportunity! Looks like it would be very straightforward to lift it into the `RunE` closure, and turning the `runDaemon` platform-hook function as a decorator for `(*daemonCLI).start()`.
```go
func runDaemon(ctx context.Context, cli *daemonCLI)
```

Reply:
Ah, yes, that's an option! I'll have a look at that in a follow-up 👍

Reply:
- opened https://github.com/moby/moby/pull/48519